transport_plugins/jlink/iotile_transport_jlink/jlink_background.py

```diff
@@ -40,7 +47,13 @@ def __init__(self, jlink_adapter, loop=SharedLoop):
         await self.write_memory(write_address, write_data, chunk_size=4)
 
         await self._trigger_rpc(device_info)
-        await self._wait_rpc_completion(control_info, timeout)
+
+        if rpc_id == self.RESET_RPC_ID:
```

Comment:
This also needs to check if `address == 8` since you can send this RPC to a peripheral tile which will not cause the `jlink` to disconnect because it just resets the tile's mcu.

Reply:
Thank you